app/src/main/java/com/example/pocketcloset/Fragments/ProfileFragment.java

```diff
@@ -121,4 +274,175 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
             }
         }
     }
+
+    public List<Clothing> filterClothing(List<Clothing> allClothing, String clothingType) {
+        ArrayList<Clothing> newClothing = new ArrayList<>();
+        for (int i = 0; i < allClothing.size(); i++) {
+            if (allClothing.get(i).getClothingType().equals(clothingType)) {
+                newClothing.add(allClothing.get(i));
+            }
+        }
+        return newClothing;
+    }
+
+    public Clothing MostWorn(String clothingType) {
+        List<Clothing> clothing = filterClothing(allClothing, clothingType);
+        if(clothing != null) {
+            Collections.sort(clothing, (d1, d2) -> {
+                return d2.getWornCount() - d1.getWornCount();
+            });
+        }
+
+        if(clothing != null && clothing.size() > 0) {
+            return clothing.get(0);
+        }
+        return null;
+    }
+
+    public Clothing LeastWorn(String clothingType) {
+        List<Clothing> clothing = filterClothing(allClothing, clothingType);
+        if(clothing != null) {
+            Collections.sort(clothing, (d1, d2) -> {
+                return d2.getWornCount() - d1.getWornCount();
+            });
+            Collections.reverse(clothing);
+        }
+
+        if(clothing != null && clothing.size() > 0) {
+            return clothing.get(0);
+        }
+        return null;
+    }
+
+
+
+
+    protected void queryMostWornClothes() {
+        // specify what type of data we want to query - Post.class
+        ParseQuery<Clothing> query = ParseQuery.getQuery(Clothing.class);
+        // include data referred by user key
+        Log.i("WardrobeFragment", "queryClothes(with params) ran");
+        query.include(Clothing.KEY_CLOTHING_IMAGE);
+        query.whereEqualTo(Clothing.KEY_USER, user);
+
+        // limit query to latest 20 items
+        query.setLimit(20);
+        // order posts by creation date (newest first)
+        query.addDescendingOrder("createdAt");
```

Comment:
Per comment above, you can set the limit to 1, sort ascending or descending by wornCount, and also filter by clothing type.